### PR TITLE
feat: handle expiration dates in idos store

### PR DESCRIPTION
**Pull request by ditoglez**

### CODE REVIEW

File: packages/idos-store/index.ts [hunk 1]
```diff
@@ -6,6 +6,18 @@ interface PipeCodecArgs<T> {
 export class Store {
   keyPrefix = "idOS-";
 
+  constructor() {
+    for (const key in localStorage) {
+      if (key.startsWith(this.keyPrefix)) {
+        const expires = localStorage.getItem(`${key}-expires`);
+        if (expires) {
+          this.reset();
```
Comment (pkoch) [inline packages/idos-store/index.ts:14]: Feels a bit too aggressive. I'd remove only this key.

File: packages/idos-store/index.ts [hunk 2]
```diff
@@ -6,6 +6,18 @@ interface PipeCodecArgs<T> {
 export class Store {
   keyPrefix = "idOS-";
 
+  constructor() {
+    for (const key in localStorage) {
+      if (key.startsWith(this.keyPrefix)) {
```
Comment (pkoch) [inline packages/idos-store/index.ts:11]: style nit:
```suggestion
      if (!key.startsWith(this.keyPrefix)) continue;
```

File: packages/idos-store/index.ts [hunk 3]
```diff
@@ -6,6 +6,30 @@ interface PipeCodecArgs<T> {
 export class Store {
   keyPrefix = "idOS-";
 
+  constructor() {
+    for (const key in localStorage) {
+      if (key.startsWith(this.keyPrefix) && key.endsWith("-expires")) {
+        const value = localStorage.getItem(key);
+        if (value) {
```
Comment (pkoch) [inline packages/idos-store/index.ts:13]:
```suggestion
        if (!value) continue;
```

File: packages/idos-store/index.ts [hunk 4]
```diff
@@ -6,6 +6,30 @@ interface PipeCodecArgs<T> {
...
+
+          try {
+            str = JSON.parse(value);
+          } catch (error) {
+            str = "";
+          }
+
+          const expires = Date.parse(str);
+
+          if (Number.isNaN(expires) || expires < Date.now()) {
```
Comment (pkoch) [inline packages/idos-store/index.ts:24]: > `Number.isNaN(expires)`

How is a Date a NaN? :thinking:

**Review by pkoch — COMMENTED** [review]
Looking pretty good! :ok_hand:

File: packages/idos-store/index.ts [hunk 5]
```diff
@@ -6,6 +6,30 @@ interface PipeCodecArgs<T> {
 export class Store {
   keyPrefix = "idOS-";
 
+  constructor() {
+    for (const key in localStorage) {
+      if (key.startsWith(this.keyPrefix) && key.endsWith("-expires")) {
```
Comment (pkoch) [inline packages/idos-store/index.ts:11]:
```suggestion
      if (!key.startsWith(this.keyPrefix) || !key.endsWith("-expires")) continue;
```

File: packages/idos-store/index.ts [hunk 6]
```diff
@@ -6,6 +6,30 @@ interface PipeCodecArgs<T> {
...
+        const value = localStorage.getItem(key);
+        if (value) {
+          let str;
+
+          try {
+            str = JSON.parse(value);
+          } catch (error) {
+            str = "";
```
Comment (pkoch) [inline packages/idos-store/index.ts:19]: Do we want to do anything else if we can't decode?
```suggestion
            continue;
```

**Comment by pkoch** [comment]
Talked with @ditoglez over a call. We're using the expiration mechanism to drive how long the enclave should remember your password field. However, we don't forget the private key, so the enclave will still happily work.

This is clearly not the intended functionality. The expiration time is not really per key, for but the whole enclave storage.

There is a question of supporting multiple human_ids being remembered. @juliosantos I'm going to make a product call to contain scope on this task. Namely, that we'll wipe the whole storage for now. If we want separate expiration dates keys on human_id, let's extract that to a different task.

File: packages/idos-store/index.ts (hunk 4 above)
Comment (pkoch) [reply packages/idos-store/index.ts]:
```js
> Date.parse("derp")
NaN
```

**Comment by pkoch** [comment]
@ditoglez :ping_pong:

**Review by sapience — APPROVED** [review]

**Pull request merged**